### PR TITLE
kubeadm: Rename FeatureFlags to FeatureGates

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:

Automatic rename from `FeatureFlags` to `FeatureGates`, as I noticed that's the real name for this feature. This is for consistency in the API and generally in the code.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```
@kubernetes/sig-cluster-lifecycle-pr-reviews @fabriziopandini @jamiehannaford

### CODE REVIEW

File: cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go [hunk 1]
```diff
@@ -52,8 +52,8 @@ type MasterConfiguration struct {
 	// UnifiedControlPlaneImage specifies if a specific container image should be used for all control plane components
 	UnifiedControlPlaneImage string `json:"unifiedControlPlaneImage"`
 
-	// FeatureFlags enabled by the user
-	FeatureFlags map[string]bool `json:"featureFlags"`
+	// FeatureGates enabled by the user
+	FeatureGates map[string]bool `json:"featureFlags"`
```
Comment (mattmoyer) [inline cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go]: This is weird to have the JSON name not match the code names.

File: cmd/kubeadm/app/apis/kubeadm/validation/validation.go [hunk 2]
```diff
@@ -297,7 +297,7 @@ func ValidateMixedArguments(flag *pflag.FlagSet) error {
 	return nil
 }
 
-func ValidateFeatureFlags(featureFlags map[string]bool, fldPath *field.Path) field.ErrorList {
+func ValidateFeatureGates(featureFlags map[string]bool, fldPath *field.Path) field.ErrorList {
```
Comment (mattmoyer) [inline cmd/kubeadm/app/apis/kubeadm/validation/validation.go]: The parameter could be renamed here as well.

File: cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go [hunk 3]
```diff
@@ -449,7 +449,7 @@ func TestValidateFeatureFlags(t *testing.T) {
 		{featureFlag{"Foo": true}, false},
 	}
 	for _, rt := range tests {
-		actual := ValidateFeatureFlags(rt.featureFlags, nil)
+		actual := ValidateFeatureGates(rt.featureFlags, nil)
 		if (len(actual) == 0) != rt.expected {
 			t.Errorf(
 				"failed featureFlags:\n\texpected: %t\n\t  actual: %t",
```
Comment (mattmoyer) [inline cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go]: The variable names and failure message in this test also.

File: cmd/kubeadm/app/cmd/phases/selfhosting.go [hunk 4]
```diff
@@ -65,7 +65,7 @@ func getSelfhostingSubCommand() *cobra.Command {
 		Short:   "Converts a Static Pod-hosted control plane into a self-hosted one.",
 		Run: func(cmd *cobra.Command, args []string) {
 			var err error
-			if cfg.FeatureFlags, err = features.NewFeatureGate(&features.InitFeatureGates, featureFlagsString); err != nil {
+			if cfg.FeatureGates, err = features.NewFeatureGate(&features.InitFeatureGates, featureFlagsString); err != nil {
```
Comment (mattmoyer) [inline cmd/kubeadm/app/cmd/phases/selfhosting.go]: `featureFlagsString` here can be renamed.

**Review by mattmoyer — COMMENTED** [review]
I noticed a few more places where we should rename for consistency. The only one that seems important is the MasterConfiguration serialized name.

**Comment by luxas** [comment]
Goh, thanks, seems like I did a case-sensitive replace ;)
Now I have both `s/FeatureFlags/FeatureGates/g;s/featureFlags/featureGates/g`

**Review by mattmoyer — APPROVED** [review]
/lgtm

**Comment by mattmoyer** [comment]
Failing test looks like https://github.com/kubernetes/kubernetes/issues/51429.

**Pull request merged**